### PR TITLE
[ie/ard] Revert to using old id

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Using the new id is [inconvenient for users](<https://github.com/yt-dlp/yt-dlp/issues/8731#issuecomment-1874398283>) bc it is just too long. Since the old id might be removed eventually though, this PR provides a fallback to the new id if it ever breaks, with a warning to open an issue about the breakage

People should use `--print-to-file "%(_old_archive_ids)#l" archive.txt` to put the new id into their archives, so when the old id disappears the connection is not completely lost

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Review by seproDev — APPROVED** [review]

**Comment by dirkf** [comment]
If the old ID really is unreliable, how about using a digest of the new ID?

**Comment by Grub4K** [comment]
Thats a fair comment, and will limit the id to a fixed length, however that means we cant get the id back from the hashed id easily. One benefit of the new ID is that we can get back to the media using that ID. Keeping the full id under `display_id` is def something we want to do either way though so that could cover that. Wed have to think about it @seproDev

File: yt_dlp/extractor/ard.py [hunk 1]
```diff
@@ -357,14 +358,25 @@ def _extract_episode_info(self, title):
...
                 'embedded': 'false',
                 'mcV6': 'true',
             })
 
+        # For user convenience we use the old contentId instead of the longer crid
+        # Ref: https://github.com/yt-dlp/yt-dlp/issues/8731#issuecomment-1874398283
+        old_id = traverse_obj(page_data, ('tracking', 'atiCustomVars', 'contentId', {str_or_none}))
+        if old_id:
+            video_id = old_id
+            archive_ids = [make_archive_id(ARDBetaMediathekIE, video_id)]
```
Comment (Grub4K) [inline yt_dlp/extractor/ard.py:374]: We have to use the new id here
```suggestion
            archive_ids = [make_archive_id(ARDBetaMediathekIE, diplay_id)]
```

File: yt_dlp/extractor/ard.py [hunk 2]
```diff
@@ -357,14 +358,25 @@ def _extract_episode_info(self, title):
...
                 'embedded': 'false',
                 'mcV6': 'true',
             })
 
+        # For user convenience we use the old contentId instead of the longer crid
+        # Ref: https://github.com/yt-dlp/yt-dlp/issues/8731#issuecomment-1874398283
+        old_id = traverse_obj(page_data, ('tracking', 'atiCustomVars', 'contentId', {str_or_none}))
+        if old_id:
+            video_id = old_id
+            archive_ids = [make_archive_id(ARDBetaMediathekIE, display_id)]
```
Comment (bashonly) [inline yt_dlp/extractor/ard.py:374]:
```suggestion
        old_id = traverse_obj(page_data, ('tracking', 'atiCustomVars', 'contentId', {int}))
        if old_id is not None:
            video_id = str(old_id)
            archive_ids = [make_archive_id(ARDBetaMediathekIE, display_id)]
```

**Review by bashonly — APPROVED** [review]

**Comment by seproDev** [comment]
We reached out to ARD and asked if the `contentId` field will be removed. Here is their response: 
> Das genannte Feld wird mittelfristig noch weiterhin ausgegeben, wird aber längerfristig ausgebaut werden. Wann genau das sein wird, können wir leider noch nicht sagen.

> This field will continue to be returned in the medium term, but will be removed in the longer term. Unfortunately, we cannot yet say exactly when this will be.

To reiterate what Grub4K said: If you are using the download archive, you should add `--print-to-file "%(_old_archive_ids)#l" archive.txt` to put the new ID into your archive. Otherwise, the connection between the old and new IDs will be completely lost once the old IDs are removed.

**Pull request merged**